extensions/icd/actions/tfu/icd.ts

```diff
@@ -41,9 +41,16 @@ export const icd: Action<
       return
     }
 
-    const accessToken = await grabAccessToken()
-
-    const icdCodes = await grabICDCodes(accessToken, icd_codes)
+    const icdCodes = await Promise.all(
+      icd_codes.map(async (code: string) => {
+        const IDPClient = makeAPIClient()
```

Comment:
you can put this constructor fn outside of the map. that way, you're reusing the token (rather than calling the auth server every time) 👍